src/ast/codegen_llvm.cpp

```diff
@@ -681,7 +681,19 @@ void CodegenLLVM::visit(Call &call)
     }
     expr_ = stackid;
   }
-
+  else if (call.func == "override_return") {
+    // int bpf_override_return(struct pt_regs *regs, u64 rc)
+    // returns: 0
+    auto &arg = *call.vargs->at(0);
+    arg.accept(*this);
+    if (arg.is_literal) {
+      b_.CreateOverrideReturn(ctx_, b_.getInt64(static_cast<Integer&>(arg).n));
```

Comment:
I'm a bit rusty on the codegen, but is it necessary to treat literals as a separate case?
Won't `arg.accept()` call into `visit(Integer &integer)`, which will then store an LLVM int64 into `expr_`? Then we'd just need `b_.CreateOverrideReturn(ctx_, expr_);` like in the "else" branch.

Reply:
yes you're right, good catch:)